kernel/arch/dreamcast/sound/snd_iface.c

```diff
@@ -136,30 +127,29 @@ void snd_sh4_to_aica_stop(void) {
    might mean a permanent failure since the queue is probably out of sync. */
 int snd_aica_to_sh4(void *packetout) {
     uint32  bot, start, stop, top, size, cnt, *pkt32;
-
-    sem_wait(&sem_qram);
+    g2_ctx_t ctx = g2_lock();
 
     /* Set these up for reference */
     bot = SPU_RAM_UNCACHED_BASE + AICA_MEM_RESP_QUEUE;
-    assert_msg(g2_read_32(bot + offsetof(aica_queue_t, valid)), "Queue is not yet valid");
+    assert_msg(g2_read_32_raw(bot + offsetof(aica_queue_t, valid)), "Queue is not yet valid");
 
-    top = SPU_RAM_UNCACHED_BASE + AICA_MEM_RESP_QUEUE + g2_read_32(bot + offsetof(aica_queue_t, size));
-    start = SPU_RAM_UNCACHED_BASE + AICA_MEM_RESP_QUEUE + g2_read_32(bot + offsetof(aica_queue_t, tail));
-    stop = SPU_RAM_UNCACHED_BASE + AICA_MEM_RESP_QUEUE + g2_read_32(bot + offsetof(aica_queue_t, head));
+    top = SPU_RAM_UNCACHED_BASE + AICA_MEM_RESP_QUEUE + g2_read_32_raw(bot + offsetof(aica_queue_t, size));
+    start = SPU_RAM_UNCACHED_BASE + AICA_MEM_RESP_QUEUE + g2_read_32_raw(bot + offsetof(aica_queue_t, tail));
+    stop = SPU_RAM_UNCACHED_BASE + AICA_MEM_RESP_QUEUE + g2_read_32_raw(bot + offsetof(aica_queue_t, head));
     cnt = 0;
-    pkt32 = (uint32 *)packetout;
+    pkt32 = (uint32_t *)packetout;
 
     /* Is there anything? */
     if(start == stop) {
-        sem_signal(&sem_qram);
+        g2_unlock(ctx);
         return 0;
     }
 
     /* Check for packet size overflow */
-    size = g2_read_32(start + offsetof(aica_cmd_t, size));
+    size = g2_read_32_raw(start + offsetof(aica_cmd_t, size));
 
-    if(cnt >= AICA_CMD_MAX_SIZE) {
-        sem_signal(&sem_qram);
+    if(size >= AICA_CMD_MAX_SIZE) {
+        g2_unlock(ctx);
```

Comment:
This changed from `cnt` to `size`, if that's on purpose, I want to know (in the commit description) why.

Reply:
Because here the packet size is checked, and not the variable `cnt` which is always zero here. Just look at the code a little more closely.